circuit.py

```diff
@@ -107,4 +114,52 @@ def merge_measurement(self, index: int) -> None:
             index (int): index of targeted rotation
         """
         pass
+
+    @staticmethod
+    def load_from_pyzx(circuit) -> 'Circuit':
+        import pyzx as zx
+
+        basic_circ = circuit.to_basic_gates()
+        ret_circ = Circuit(basic_circ.qubits, input_circ.name)
+
+
+        for gate in basic_circ_gates:
+            gate_missed = 0
+            if isinstance(gate, zx.circuit.ZPhase):
+                pauli_rot = gate.phase / 2
```

Comment:
You can use this function to break angles in their components:
```
from fractions import Fraction
from typing import *
import math

def decompose_pi_fraction(f:Fraction) -> List[Fraction]:
  """
  Break a fraction that has a power of two as denominator into a sequence of fractions that add up to it, all of which have 1 for numerator.

  Adjusts the fraction to be in the (2,0] range
  """

  # Check that f's denominator is a power of two
  assert(f.denominator & (f.denominator - 1) == 0)

  # Adjust to the the (2*pi,0] range
  f = Fraction(f.numerator % (2*f.denominator), f.denominator)

  if f.numerator == 0: return [f]
  elif f.numerator == 1: return [f]
  else:
      p = int(math.log(f.numerator, 2))
      largest_pow_of_2 = int(pow(2, p))  
      return [Fraction(largest_pow_of_2, f.denominator)] \
        + decompose_pi_fraction(Fraction(f.numerator - largest_pow_of_2,f.denominator))
```
Example:
```
print(decompose_pi_fraction(Fraction(0,4  )))
print(decompose_pi_fraction(Fraction(1,2  )))
print(decompose_pi_fraction(Fraction(1,4  )))
print(decompose_pi_fraction(Fraction(7,4  )))
print(decompose_pi_fraction(Fraction(9,4  )))
print(decompose_pi_fraction(Fraction(-1,8 )))
print(decompose_pi_fraction(Fraction(15,16)))
print(decompose_pi_fraction(Fraction(-3,4 )))
```
produces
```
[Fraction(0, 1)]
[Fraction(1, 2)]
[Fraction(1, 4)]
[Fraction(1, 1), Fraction(1, 2), Fraction(1, 4)]
[Fraction(1, 4)]
[Fraction(1, 1), Fraction(1, 2), Fraction(1, 4), Fraction(1, 8)]
[Fraction(1, 2), Fraction(1, 4), Fraction(1, 8), Fraction(1, 16)]
[Fraction(1, 1), Fraction(1, 4)]
```

Reply:
Also
```
print(sum(decompose_pi_fraction(Fraction(1,2  ))))
print(sum(decompose_pi_fraction(Fraction(1,4  ))))
print(sum(decompose_pi_fraction(Fraction(7,4  ))))
print(sum(decompose_pi_fraction(Fraction(9,4  ))))
print(sum(decompose_pi_fraction(Fraction(-1,8 ))))
print(sum(decompose_pi_fraction(Fraction(15,16))))
print(sum(decompose_pi_fraction(Fraction(-3,4 ))))
```
produces
```
0
1/2
1/4
7/4
1/4
15/8
15/16
```